test/integration/load_stats_integration_test.cc

```diff
@@ -364,7 +365,8 @@ TEST_P(LoadStatsIntegrationTest, Success) {
   }
 
   // Verify we do not get empty stats for non-zero priorities.
-  waitForLoadStatsRequest({localityStats("winter", 2, 0, 0), localityStats("dragon", 2, 0, 0)});
+  waitForLoadStatsRequest(
```

Comment:
Take a look at the `waitForLoadStatsRequest` implementation, I think this needs some work to avoid the coverage hang. It works on multiple samples until a result is established, the `total_issued` stats are only for last sample.

Reply:
waitForLoadStatsRequest is the most likely suspect for the coverage hang. I did not make progress by looking at the test code. Wondering if we could enable the debug logs(per test case) when the coverage test runs?  IMO this should greatly help. Any pointers on how to do this? @htuch @lizan . I tried a bunch of hacks in run_envoy_bazel_coverage.sh but no success so far. I always get the gtest pass/fail status output but never the test logs.

FYI - Looks like the test "Success" passes. It the next test test "LocalityWeighted" which hangs. Locally all tests pass. IMO this points to a timing issue. Logs should help.

Reply:
triggered a coverage run with logging...

Reply:
Ping me on IM re: how to get artifacts out of CI if you want. I think it races because we're essentially trying to wait on a fixed quantity which only is valid in the first loop around.

Reply:
Coverage started passing as soon as i added some logs. This points to a race but unable to figure out where the race is.

IIUC - you're saying that there is a race between calling WaitForLoadStatsRequest and when the load stats message actually arrives. 

I did a quick test of this hypothesis:
I added an arbitrary sleep(10 sec) before calling WaitForLoadStatsRequest. I would've expected the test to timeout, but instead the test passes. 
WaitForLoadStatsRequest receives the expected load stats in the first message that we receive after waiting for 10secs (> load report interval of 500msec). Verified that the test indeed waits for 10secs through the logs.
https://github.com/envoyproxy/envoy/pull/6784/files#diff-c28c582462ec27cf091a0eac2e84fe4eR490
Looks like load stats messages are being read by WaitForLoadStatsRequest in order.

I'm going to trigger the coverage run again without any of the logging and sleeps. Maybe something got fixed somewhere else now.

Reply:
Have you updated https://github.com/envoyproxy/envoy/blob/d236641d328557d1ec4074c39c7a06abf963a49a/test/integration/load_stats_integration_test.cc#L191 to handle your new field?

Reply:
Oops. this fixes the issue. Thanks for the catch Harvey!